Fix Code region remains highlighted when using command `Unfold All Lines`

Nirhar opened the pull request:
Previously the handler for `Unfold All Lines` command only executed `_unhide_all_lines` which does not remove the highlighting associated with a code region. This has been replaced with a call to `unfold_line` for every line, which also removes highlighting along with unhiding the lines.

Fixes #83979

AThousandShips commented:
Your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

Nirhar committed:
Fix Code region remains highlighted when using command `Unfold All Lines`

Previously the handler for `Unfold All Lines` command only
executed `_unhide_all_lines` which does not remove the
highlighting associated with a code region. This has been
replaced with a call to `unfold_line` for every line, which
also removes highlighting along with unhiding the lines.

Fixes #83979

Nirhar commented:
Corrected it, thanks for pointing it out!

Nirhar commented:
@kitbdev @tokengamedev based on your discussion in issue #83979 , I think it is sufficient to just visit every line using a loop and unfold it.

kitbdev commented on scene/gui/code_edit.cpp:
```diff
@@ -1734,7 +1734,10 @@ void CodeEdit::fold_all_lines() {
+	for (int i = 0; i < get_line_count(); i++) {
+		unfold_line(i);
+	}
+	queue_redraw();
```
You shouldn't need to `queue_redraw` here since `unfold_line` does handle it.

`queue_redraw` puts the `CanvasItem` in the queue, see this for details: https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#class-canvasitem-method-queue-redraw

YuriSizov replied:
`CodeEdit::fold_all_lines()` has the same issue, though it ultimately doesn't matter. The items is only queued once (otherwise doing it in the loop via `(un)fold_line` would be a problem as well).

Nirhar replied:
Removed the `queue_redraw` call at the end of `unfold_all_lines`

Paulb23 reviewed (commented):
Thanks for the contribution!

Looks like this would need to be handled in `set_line_folding_enabled`, as it also calls `_unhide_all_lines`, which would case the same issue.

At that point `_unhide_all_lines` would also be unused so you could remove it.

Would also be good to ensure this behaviour with a couple unit tests.

YuriSizov commented:
> Looks like this would need to be handled in `set_line_folding_enabled`, as it also calls `_unhide_all_lines`, which would case the same issue.
> 
> At that point `_unhide_all_lines` would also be unused so you could remove it.

@Nirhar Would you be around to address this part? Adding new unit tests would also be good, but not critical to get this merged.

kitbdev commented:
- Superseded by https://github.com/godotengine/godot/pull/106441

Thanks for the contribution nonetheless!
